core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -641,8 +641,12 @@ package object matryoshka {
     *
     * @group algtrans
     */
-  def repeatedly[A](f: A => Option[A]): A => A =
-    expr => f(expr).fold(expr)(repeatedly(f))
+  @annotation.tailrec
```

Comment:
We use a custom Predef, which provides `tailrec` (but doesn’t include `_root_.scala`), so you don’t need to qualify it.